### PR TITLE
Use snappy compression for digests

### PR DESCRIPTION
**Pull request by dgnorton**
This PR backports https://github.com/influxdata/influxdb/pull/10215 to 1.6.

### CODE REVIEW

File: tsdb/engine/tsm1/digest.go [hunk 1]
```diff
@@ -142,6 +142,7 @@ func DigestFresh(dir string, files []string, shardLastMod time.Time) (bool, stri
 	if err != nil {
 		return false, fmt.Sprintf("Can't read digest: err=%s", err)
 	}
+	defer dr.Close()
```
Comment (jacobmarble) [inline tsdb/engine/tsm1/digest.go]: `defer` is good to ensure cleanup on early return, but can you also check the error returned by `dr`, perhaps just after the last usage? `Close` should be idempotent.

File: tsdb/engine/tsm1/digest_reader.go [hunk 2]
```diff
@@ -1,84 +1,73 @@
...
 
 	return string(b), ts, nil
 }
 
-func (w *DigestReader) Close() error {
-	if err := w.gr.Close(); err != nil {
-		return err
-	}
-	return w.r.Close()
+func (r *DigestReader) Close() error {
```
Comment (jacobmarble) [inline tsdb/engine/tsm1/digest_reader.go]: No need to close `r.sr`?

Comment (dgnorton) [reply tsdb/engine/tsm1/digest_reader.go]: Surprised me, too, but `snappy.Reader` doesn't have a `Close`. https://github.com/golang/snappy/blob/master/decode.go

File: tsdb/engine/tsm1/digest.go (hunk 1 above)
Comment (dgnorton) [reply tsdb/engine/tsm1/digest.go]: Good idea. Since this is a backport of something on `master`, I'll make a note and change it on a future commit.

**Pull request merged**